File: Shanjay K/Lab 2/javascript/javascript_playground/kafka/consumer.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")
const msg = process.argv[2];

run();
async function run(){
    try{
        const kafka = new Kafka({
            "cliendId": "myapp",
            "brokers": ["localhost:9093"]
        })

        const consumer = kafka.consumer({"groupId": "test"});
        console.log("Connecting...")
        await consumer.connect()
        console.log("Connected!")
        
        consumer.subscribe({
            "topic": "Users",
            "fromBeginning": true
        })

        await consumer.run({
            "eachMessage": async result => {
                console.log(`Received message: ${result.message.value} on partition ${result.partition}`)
            }
        })
        //await consumer.disconnect();

    }
    catch(ex)
    {
        console.error(`Error: ${ex}`)
    }
    finally {
        
    }
}

File: Yusuf Shaik/Kafka Python/consumer.py
from kafka import KafkaConsumer
import sys

groupId = sys.argv[1]
consumer = KafkaConsumer(
    'Users', 
    group_id = groupId, 
    bootstrap_servers=['localhost:9093', 'localhost:9094', 'localhost:9095'])

print("Ready")

for message in consumer:
    print ("Topic = %s:%d:%d:  value=%s partition=%s" % (message.topic, message.partition,
                                          message.offset,
                                          message.value, message.partition))


    